src/frontends/onnx/frontend/src/op/cum_sum.cpp

```diff
@@ -20,7 +20,12 @@ OutputVector cum_sum(const Node& node) {
     Output<ngraph::Node> axis;
 
     if (inputs.size() > 1) {
-        axis = inputs.at(1);  // optional input, 0-D tensor
+        // optional input, 0-D or 1-D tensor
+        const auto& axis_shape = inputs.at(1).get_partial_shape();
+        NGRAPH_CHECK(axis_shape.rank().get_length() == 0 || axis_shape.rank().get_length() == 1,
+                     "axis shape should be 0-D or 1-D.");
+        axis =
```

Comment:
Although ONNX only allows scalar axis input I think this is ok. I'd like to see it demonstrated with even a single test though. Please add it to this file https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp#L621 and use any of the existing CumSum tests as a template.